### PR TITLE
test(step-bar): improve testing

### PR DESCRIPTION
**Pull request by domw30**
- improve testing of step bar
- increase testing coverage of step bar

### CODE REVIEW

File: src/components/StepBar/StepBar.test.tsx [hunk 1]
```diff
@@ -1,58 +1,123 @@
...
+
+      expect(mockOnChangeStep).toHaveBeenCalledTimes(2);
+    });
+
+    test('should not call onChangeStep when undefined and clicked', () => {
+      render(<StepBar {...DEFAULT_PROPS} onChangeStep={() => undefined} />);
+
+      fireEvent.click(screen.getByText(MOCK_STEPS[1].title));
+      expect(mockOnChangeStep).not.toHaveBeenCalled();
+    });
```
Comment (colbr) [inline src/components/StepBar/StepBar.test.tsx:92]: I think we can omit this one - the component will never call a function it isn't passed.

File: src/components/StepBar/StepBar.test.tsx [hunk 2]
```diff
@@ -1,58 +1,123 @@
...
 
-  expect(mockOnChangeStep).toHaveBeenCalledWith(mockSteps[0]);
+      expect(currentStep[0].textContent).toBe(MOCK_STEPS[2].title);
+      expect(currentStep[0].textContent).not.toBe(MOCK_STEPS[1].title);
+      expect(currentStep[0].textContent).not.toBe(MOCK_STEPS[0].title);
+    });
+
+    test('should apply Hide styles when current step index is greater than number of steps', () => {
+      const { container } = render(
+        <StepBar {...DEFAULT_PROPS} currentStepId={MOCK_STEPS[2].id} steps={[MOCK_STEPS[0]]} />
```
Comment (colbr) [inline src/components/StepBar/StepBar.test.tsx:116]: Hmm perhaps we can improve types to prevent this - I'll have a think about it!

**Review by colbr — CHANGES_REQUESTED** [review]
A few tiny changes, but otherwise looks perfect. Nice work!

**Comment by colbr** [comment]
Sorry for switching PR target a couple times! I was thinking we should merge all of these testing changes into one branch, then bring them over all at once, but that's just going to be annoying, so we'll just go to `master`.

File: src/components/StepBar/StepBar.test.tsx [hunk 3]
```diff
@@ -1,58 +1,123 @@
...
-  const { getByText } = renderComponent({
-    currentStepId: mockSteps[2].id,
-    steps: mockSteps,
-    onChangeStep: mockOnChangeStep
+  describe('step', () => {
+    test('should apply the step title', () => {
+      render(<StepBar {...DEFAULT_PROPS} currentStepId={MOCK_STEPS[1].id} />);
+
+      expect(screen.getByText(MOCK_STEPS[0].title)).toBeInTheDocument();
+    });
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/components/StepBar/StepBar.test.tsx:66]: Ah yes, I tweaked the name of test slightly to highlight this
`should render each step and in the correct order and apply titles`

File: src/components/StepBar/StepBar.test.tsx [hunk 4]
```diff
@@ -1,58 +1,123 @@
...
+
+    test('should contain ArrowLink', () => {
+      const { container } = render(<StepBar {...DEFAULT_PROPS} />);
+      const step = container.getElementsByClassName(styles.Step);
+
+      expect(step[0].firstChild).toHaveClass('ArrowLink');
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/components/StepBar/StepBar.test.tsx:72]: Sweet, think I've improved this well ... Updated.

File: src/components/StepBar/StepBar.test.tsx (hunk 1 above)
Comment (domw30) [reply src/components/StepBar/StepBar.test.tsx:92]: Makes sense!

**Pull request closed without merging**